File: src/main/scala/org/xbery/artbeams/users/service/LoginService.scala
package org.xbery.artbeams.users.service

import javax.servlet.http.HttpServletRequest
import org.xbery.artbeams.users.domain.User

/**
  * User login operations.
  * @author Radek Beran
  */
trait LoginService {

  /**
    * Returns user object if user should be logged.
    */
  def login(username: String, password: String): Option[User]

  /**
    * Returns logged user or None.
    * @param request
    * @return
    */
  def getLoggedUser(request: HttpServletRequest): Option[User]
}

File: src/main/kotlin/org/xbery/artbeams/users/service/LoginService.kt
package org.xbery.artbeams.users.service

import jakarta.servlet.http.HttpServletRequest
import org.xbery.artbeams.common.error.requireAuthenticated
import org.xbery.artbeams.users.domain.User

/**
 * User login operations.
 * @author Radek Beran
 */
interface LoginService {
    fun getLoggedUser(request: HttpServletRequest): User?

    fun requireLoggedUser(request: HttpServletRequest): User {
        return requireAuthenticated(getLoggedUser(request)) {
            "User must be logged in"
        }
    }

    /**
     * Logs the user in.
     */
    fun loginUser(user: User, plainPassword: String, request: HttpServletRequest)
}
